docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,207 @@
+It is is a list of strings, each of which is a single Rule Entry that corresponds to a single HAP Pool. 
+Each Rule Entry is evaluated during a cluster provisioning process.
+If more than one rule applies the best matching rule is selected based on the rule [priority](#priority). The following examples presents the Rule Configuration in values.yaml:
+
+```
+hap: 
+  rule: 
+  - rule_1
+  - rule_2
+  ...
+  - rule_n
+```
+
+Every HAP Pool that the Rule Entry represents must be preconfigured in a Gardener Cluster that KEB connects to.
+
+## Rule Format 
+
+The following example shows the Rule Entry format:
+
+```
+PLAN(ATTR_1=VAL_1,ATTR_2=VAL_2,...,ATTR_N)
+```
+
+In its minimal form each rule consists of a `PLAN` that the rule applies to.
+The rule can be extended with a list of attributes and its values passed in the form of `ATTR=VAL` pairs.
+Description of when the rule is triggered can be found in the [Rule Evaluation](#evaluation) section.
+List of possible attributes is described in the [Rule Attributes](#rule-attributes) section.
+
+```
+hap: 
+  rule: 
+  - azure(ATTR_1=VAL_1,ATTR_2=VAL_2,...,ATTR_N)
+```
+
+The extended version of a rule entry allows to pass a list of attribute/value pairs in parantheses. The attributes include **platformRegion** (`PR`), **clusterRegion** (`CR`), **shared** (`S`) and **euAccess** (`E`). Each attribute can be used at most once in a single rule entry.
+shared and euAccess attributes are logical which means that they are not matched with SKR attributes but are used only to apply their labels modifications if rest of matched attributes are equal to SKR's values. 
+
+Rule Attributes are described in [Rule Attributes](#rule-attributes) section.
+
+## Rule Evaluation
+
+HAP evaluates a set of rules to determine what labels to use when querying secret bindings.
+Rule is evaluated during cluster provisioning.
+Rule entries are analyzed one by one.
+Physical attributes from every entry are compared with values from input, if they are the same the rule entry is considered triggered.
+If there is more than one triggered rule then only one is selected in accordance to priority rules described in the [Priority](#priority) section.
+If there are no rule entries trigered then the error is returned. No fallback behaviour is defined in such case.
+
+## Priority
+
+Only one rule can be triggered. If more than one rule entry match the request than only one is selected by sorting them by the number of attributes they contain.
+The entry that specifies most attributes (hence is more specific) is selected.
+If there are more than one rule with the same number of attributes then an error is returned.
+
+## Search Labels
+
+The HAP stores credentials for the hyperscaler accounts that have been set up in advance in Kubernetes Secrets that the Secret Bindings, that KEB searches for, point to. The SecretBindings are labeld with **hyperscalerType**, **shared** and **euAccess** labels. 
```

Comment:
this must fail:

1)
Rule:
- gcp
- gcp(S)

2)
Rule:
- gcp(PR=cf-eu11)
- gcp(S, PR=cf-eu11)

3)
Rule:
- gcp(PR=cf-eu11)
- gcp(S, PR=cf-eu11, euAccess)